File: ruby/abc106_b.rb
# 2
# ABC106/B
# 全探索：全列挙

N = gets.to_i
divisors = [0] * (N + 1)

1.step(N, 2) do |i|
  i.step(N, i) { divisors[_1] += 1 }
end

puts 1.step(N, 2).count { divisors[_1] == 8 }

File: atcoder_selected_100/kotlin/abc106_b.kt
/** 2
 * ABC106/B
 * 全探索:全列挙
 */

 fun main() {
  val N = readLine()!!.toInt()
  val divisor = MutableList(N + 1) { 0 }

  (1 .. N step 2).forEach { i ->
    (i .. N step i).forEach { divisor[it]++ }
  }

  println((1 .. N step 2).count { divisor[it] == 8 })
}
